transport/pool_integration_test.go

```diff
@@ -10,7 +10,7 @@ import (
 
 const refillerBackoff = 250 * time.Millisecond
 
-func TestConnPoolIntegration(t *testing.T) {
```

Comment:
Naming can be improved.
What you have here is more often called newTestConnPool.

Typically a test helper it is a struct of its own, for example containing a pool as a field - not needed here IMO.